socialdistribution/api/utils.py

```diff
@@ -0,0 +1,44 @@
+from profiles.models import Author
+from posts.models import Post, Comment
+
+
+def post_to_dict(post):
+    comments = Comment.objects.filter(post=post)
+
+    return {
+        "title" : post.title,
+        "source" : "POST HAS NO ATTRIBUTE SOURCE",
+        "origin" : "POST HAS NO ATTRIBUTE ORIGIN",
+        "description" : post.description,
+        "contentType" : post.content_type,
+        "content" : "POST HAS NO ATTRIBUTE CONTENT",
+        "author" : author_to_dict(post.author),
+        "categories":["web","tutorial"],
+        "count" : comments.count(),
+        "size" : "IMPLEMENT PAGINATION",
+        "next" : "IMPLEMENT PAGINATION",
+        "comments" : [comment_to_dict(comment) for comment in comments],
+        "published" : post.published,
+        "id" : post.id,
+        "visibility" : post.visibility,
+        "visibleTo" : post.visibileTo,
+        "unlisted" : post.unlisted
```

Comment:
Should the post api handle posting images as well?

Reply:
I think should have the image for api as well. To the image location?

Reply:
An image is a post. The content type would be image, and the content would be the image in binary.

Reply:
Oops, I have added an Imagefield to the post model in order to host the image in a folder called /media. : ( So in this case, we will have to encoded the image and put the binary in content?